File: backend/internal/db/tables/recipe.go
package tables

import "github.com/google/uuid"

type Recipe struct {
	ID             uuid.UUID
	Name           string
	UniqueName     string
	Description    string
	OvenTemp       int
	EstimatedTime  int
	Deleted        bool
	CreatedBy      uuid.UUID
	Portions       int
	PortionsSuffix string
}

func (_ Recipe) StructName() string {
	return "Recipe"
}

File: frontend/src/api/Recipe.ts
import { Author } from "./Author";
import { Image } from "./Image";
import { Ingredient } from "./Ingredient";
import { Step } from "./Step";
import { Tag } from "./Tag";

export interface Recipe {
  id: string;
  name: string;
  uniqueName: string;
  description: string;
  ovenTemperature: number;
  estimatedTime: number;
  steps: Step[];
  ingredients: Ingredient[];
  images: Image[];
  author: Author;
  tags: Tag[];
  portions: number;
  portionsSuffix: string;
}
